google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -368,13 +368,107 @@ public void defaultQuery() throws Exception {
     assertEquals("bar", documents.next().get("foo"));
   }
 
+  @Test
+  public void defaultQueryStream() throws Exception {
+    addDocument("foo", "bar");
+    addDocument("foo", "bar");
+
+    final Semaphore semaphore = new Semaphore(0);
+    final Iterator<String> iterator = Arrays.asList("bar", "bar").iterator();
+    randomColl.stream(
+        new ApiStreamObserver<DocumentSnapshot>() {
+          @Override
+          public void onNext(DocumentSnapshot documentSnapshot) {
+            assertEquals(iterator.next(), documentSnapshot.get("foo"));
+          }
+
+          @Override
+          public void onError(Throwable throwable) {}
+
+          @Override
+          public void onCompleted() {
+            semaphore.release();
+          }
+        });
+
+    semaphore.acquire();
+    // Verify the number of response
+    assertFalse(iterator.hasNext());
+  }
+
+  @Test
+  public void largeQuery() throws Exception {
+    int count = 100;
+    while (count-- > 0) {
+      addDocument("foo", "bar");
+    }
+
+    QuerySnapshot querySnapshot = randomColl.get().get();
+    assertEquals(100, querySnapshot.size());
+  }
+
+  @Test
+  public void largeQueryStream() throws Exception {
+    int count = 100;
+    while (count-- > 0) {
+      addDocument("foo", "bar");
+    }
+
+    final Semaphore semaphore = new Semaphore(0);
+    final int[] docCount = {0};
+    randomColl.stream(
+        new ApiStreamObserver<DocumentSnapshot>() {
+          @Override
+          public void onNext(DocumentSnapshot documentSnapshot) {
+            docCount[0]++;
+          }
+
+          @Override
+          public void onError(Throwable throwable) {}
```

Comment:
Consider adding assertions within the onError callback to catch any unexpected exceptions